tests/db_engine_specs/pinot_tests.py

```diff
@@ -32,10 +32,48 @@ def test_pinot_time_expression_sec_one_1d_grain(self):
             "DATETIMECONVERT(tstamp, '1:SECONDS:EPOCH', '1:SECONDS:EPOCH', '1:DAYS')",
         )  # noqa
 
+    def test_pinot_time_expression_simple_date_format_1d_grain(self):
+        col = column("tstamp")
+        expr = PinotEngineSpec.get_timestamp_expr(col, "%Y-%m-%d %H:%M:%S", "P1D")
+        result = str(expr.compile())
+        self.assertEqual(
+            result,
+            (
+                "DATETIMECONVERT(tstamp, "
+                + "'1:SECONDS:SIMPLE_DATE_FORMAT:yyyy-MM-dd HH:mm:ss', "
+                + "'1:SECONDS:SIMPLE_DATE_FORMAT:yyyy-MM-dd HH:mm:ss', '1:DAYS')"
+            ),
+        )  # noqa
+
+    def test_pinot_time_expression_simple_date_format_1w_grain(self):
+        col = column("tstamp")
+        expr = PinotEngineSpec.get_timestamp_expr(col, "%Y-%m-%d %H:%M:%S", "P1W")
+        result = str(expr.compile())
+        self.assertEqual(
+            result,
+            (
+                "ToDateTime(DATETRUNC('week', FromDateTime(tstamp, "
+                + "'yyyy-MM-dd HH:mm:ss'), 'MILLISECONDS'), 'yyyy-MM-dd HH:mm:ss')"
+            ),
+        )  # noqa
+
     def test_pinot_time_expression_sec_one_1m_grain(self):
         col = column("tstamp")
         expr = PinotEngineSpec.get_timestamp_expr(col, "epoch_s", "P1M")
         result = str(expr.compile())
         self.assertEqual(
             result, "DATETRUNC('month', tstamp, 'SECONDS')",
         )  # noqa
+
+    def test_invalid_get_time_expression_arguments(self):
+        with self.assertRaises(NotImplementedError) as context:
+            PinotEngineSpec.get_timestamp_expr(column("tstamp"), None, "P1M")
+        self.assertEqual("Empty date format for 'tstamp'", str(context.exception))
```

Comment:
Is this indented correctly? I'm fine removing the string comparison, testing for the raise is adequate IMO.

Reply:
I tried to use VS Code pylint plugin and black to format the python file, but both don't pass the python lint check here :(
The black cmd I'm running is 
```black -t py37 tests/db_engine_specs/pinot_tests.py```

Reply:
I also tried:
``` self.assertRaises(
            NotImplementedError,
           PinotEngineSpec.get_timestamp_expr(column("tstamp"), None, "P1M"),
        )
```
but the test doesn't pass for this, hence I capture the exception and compare the string:
```
================================================================================================= FAILURES ==================================================================================================
_____________________________________________________________________ TestPinotDbEngineSpec.test_invalid_get_time_expression_arguments ______________________________________________________________________

self = <tests.db_engine_specs.pinot_tests.TestPinotDbEngineSpec testMethod=test_invalid_get_time_expression_arguments>

    def test_invalid_get_time_expression_arguments(self):
        self.assertRaises(
            NotImplementedError,
>           PinotEngineSpec.get_timestamp_expr(column("tstamp"), None, "P1M"),
        )

tests/db_engine_specs/pinot_tests.py:71:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'superset.db_engine_specs.pinot.PinotEngineSpec'>, col = <sqlalchemy.sql.elements.ColumnClause at 0x11f54a5e0; tstamp>, pdf = None, time_grain = 'P1M', type_ = None

    @classmethod
    def get_timestamp_expr(
        cls,
        col: ColumnClause,
        pdf: Optional[str],
        time_grain: Optional[str],
        type_: Optional[str] = None,
    ) -> TimestampExpression:
        if not pdf:
>           raise NotImplementedError(f"Empty date format for '{col}'")
E           NotImplementedError: Empty date format for 'tstamp'

superset/db_engine_specs/pinot.py:72: NotImplementedError
```

Reply:
I tweaked the tests slightly and pushed to the branch, I hope you don't mind (if you don't agree with the changes please feel free to `git reset --hard HEAD~1`!)